packages/mco/components/data-policies/create-dr-policy/create-dr-policy.tsx

```diff
@@ -138,31 +222,142 @@ type SyncScheduleProps = {
   setSchedule: React.Dispatch<React.SetStateAction<string>>;
 };
 
-export const CreateDRPolicy: React.FC<{}> = ({}) => {
+type ReRouteResourceProps = {
+  history: RouteComponentProps['history'];
+  match: Match<{ url: string }>;
+};
+
+type MirrorPeerMapping = {
+  [name in string]: MirrorPeerKind;
+};
+
+export const CreateDRPolicy: React.FC<ReRouteResourceProps> = ({match, history}) => {
   const { t } = useTranslation('plugin__odf-console');
-  const history = useHistory();
+  const { url }  = match;
+  
   const [policyName, setPolicyName] = React.useState('');
-  const [replication] = React.useState('');
-  const [selectedClusters, setSelectedClusters] = React.useState<Cluster[]>([]);
+  const [replication, setReplication] = React.useState('');
+  const [manageS3Profile, setManageS3Profile] = React.useState(false);
   const [syncTime, setSyncTime] = React.useState('5m');
+  const [isOCSDetected, setOCSDetected] = React.useState(false);
+  const [selectedClusters, setSelectedClusters] = React.useState<{[name in string] : Cluster}>({});
+  const [isReplicationOpen, setReplicationOpen] = React.useState(false);
+  const [isReplicationInputManual, setReplicationInputManual] = React.useState(false);
+  const [errorMessage, setError] = React.useState("");
+  const [mirrorPeerMapping, setMirrorPeerMapping] = React.useState<MirrorPeerMapping>({});
+
+
+  const [mirrorPeers, mirrorPeerLoaded, mirrorPeerLoadError] = useK8sWatchResource<MirrorPeerKind[]>({
+    kind: referenceForModel(MirrorPeerModel),
+    isList: true,
+    namespaced: false,
+  });
+
+  React.useEffect(() => {
+    setMirrorPeerMapping(mirrorPeers?.reduce((obj, mirrorPeer) => ({...obj, [mirrorPeer?.metadata?.name] : mirrorPeer}), {}));
+  }, [mirrorPeers, mirrorPeerLoaded, mirrorPeerLoadError]);
+
+  React.useEffect(() => {
+    if (Object.keys(selectedClusters)?.length === 2) {
+      // OCS detection
+      setOCSDetected(Object.values(selectedClusters)?.reduce((acc, cluster) => 
+        (acc && cluster?.storageSystem !== "" && cluster?.isValidODFVersion), true));
+
+      let storageClusterId = new Set('');
+      // DR replication type
+      Object.values(selectedClusters)?.every((cluster) => {
+        if (cluster?.storageClusterId !== "") {
+          storageClusterId.add(cluster?.storageClusterId);
+          return true;
+        };
```

Comment:
Use "reduce" its much cleaner 
```suggestion
     const storageClusterIDs = Object.values(selectedClusters)?.reduce((ids, cluster) => {
        if (cluster?.storageClusterId !== "") {
          ids.add(cluster?.storageClusterId);
        };
        return ids;
        }, new Set(''));
```

Reply:
reduce will loop through all the element, here i want to hard stop the looping and assinging  setReplicationInputManual(true);

here, we dont need to loop further even one selected cluster dont have storageClusterId. i feel every is the best this usecase.

Reply:
Okay,  I missed those setStates below in previous review thinking its outside the loop. if thats the case then you can do something like this as well:
```suggestion
    const clustersData = Object.values(selectedClusters ?? []);
   
      const hasReplicationInputManual = clustersData?.every((cluster) => clustersData?.storageClusterId !== "");
    
      const storageClusterIDs = clustersData?.reduce((ids, cluster) => {
      if (cluster?.storageClusterId !== "") {
	          ids.add(cluster?.storageClusterId);
	        };
	        return ids;
      }, new Set(''));
      
setReplicationInputManual(hasReplicationInputManual);
setReplication(hasReplicationInputManual || storageClusterId.size > 1 ? REPLICATION_TYPE(t)['async'] : REPLICATION_TYPE(t)['sync']);
```

Currently the returned value from every is not utilized anywhere.